File: androidApp/src/main/java/com/aitrainingapp/android/ui/profile/ProfileScreen.kt
package com.aitrainingapp.android.ui.profile

import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import coil.compose.rememberAsyncImagePainter
import com.aitrainingapp.android.room.ProfileEntity
import com.aitrainingapp.android.viewmodel.ProfileViewModel
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ProfileScreen(viewModel: ProfileViewModel) {
    val profiles by viewModel.profiles.collectAsState()
    val username by viewModel.username.collectAsState()
    val aiId by viewModel.aiIdentifier.collectAsState()
    val selectedId by viewModel.selectedProfileId.collectAsState()

    LaunchedEffect(Unit) {
        viewModel.loadData()
    }

    var expanded by remember { mutableStateOf(false) }
    var selectedProfile by remember { mutableStateOf<ProfileEntity?>(null) }

    LaunchedEffect(profiles, selectedId) {
        if (selectedProfile == null && profiles.isNotEmpty()) {
            selectedProfile = profiles.find { it.id == selectedId } ?: profiles.first()
        }
    }

    val coroutineScope = rememberCoroutineScope()

    Surface(
        modifier = Modifier.fillMaxSize(),
        color = MaterialTheme.colorScheme.background
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(24.dp)
        ) {
            Spacer(modifier = Modifier.height(20.dp))

            Text(
                text = "Profil użytkownika",
                style = MaterialTheme.typography.headlineMedium,
                color = MaterialTheme.colorScheme.onBackground
            )

            Spacer(modifier = Modifier.height(24.dp))

            val avatarUrl = remember {
                val id = (1..1000).random()
                "https://api.dicebear.com/7.x/personas/png?seed=user$id"
            }

            Image(
                painter = rememberAsyncImagePainter(avatarUrl),
                contentDescription = "User Avatar",
                modifier = Modifier
                    .size(100.dp)
                    .clip(CircleShape)
                    .align(Alignment.CenterHorizontally)
            )

            Spacer(modifier = Modifier.height(16.dp))

            Text(
                text = username,
                style = MaterialTheme.typography.titleLarge,
                color = MaterialTheme.colorScheme.onBackground,
                modifier = Modifier.align(Alignment.CenterHorizontally)
            )
            Text(
                text = aiId,
                style = MaterialTheme.typography.bodyMedium,
                color = Color.LightGray,
                modifier = Modifier.align(Alignment.CenterHorizontally)
            )

            Spacer(modifier = Modifier.height(32.dp))

            Text(
                text = "Wybierz profil",
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onBackground
            )

            Spacer(modifier = Modifier.height(8.dp))

            ExposedDropdownMenuBox(
                expanded = expanded,
                onExpandedChange = { expanded = !expanded }
            ) {
                OutlinedTextField(
                    value = selectedProfile?.name ?: "",
                    onValueChange = {},
                    readOnly = true,
                    label = { Text("Profil") },
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = expanded) },
                    modifier = Modifier
                        .menuAnchor()
                        .fillMaxWidth(),
                    colors = OutlinedTextFieldDefaults.colors(
                        unfocusedBorderColor = Color.Gray,
                        focusedBorderColor = MaterialTheme.colorScheme.primary,
                        unfocusedTextColor = MaterialTheme.colorScheme.onBackground,
                        focusedTextColor = MaterialTheme.colorScheme.onBackground,
                        focusedLabelColor = MaterialTheme.colorScheme.primary,
                        unfocusedLabelColor = Color.Gray,
                        cursorColor = MaterialTheme.colorScheme.primary
                    )
                )

                ExposedDropdownMenu(
                    expanded = expanded,
                    onDismissRequest = { expanded = false },
                    modifier = Modifier.exposedDropdownSize()
                ) {
                    profiles.forEach { profile ->
                        DropdownMenuItem(
                            text = {
                                Text(
                                    profile.name,
                                    color = MaterialTheme.colorScheme.onSurface
                                )
                            },
                            onClick = {
                                selectedProfile = profile
                                expanded = false
                            },
                            colors = MenuDefaults.itemColors(
                                textColor = MaterialTheme.colorScheme.onSurface,
                                disabledTextColor = Color.Gray
                            )
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(24.dp))

            selectedProfile?.let { profile ->
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 8.dp)
                ) {
                    Text(
                        text = "Szczegóły profilu:",
                        style = MaterialTheme.typography.titleMedium,
                        color = MaterialTheme.colorScheme.primary
                    )
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        text = "Zmiana ciężaru: ${profile.weightChance}kg",
                        color = MaterialTheme.colorScheme.onBackground,
                        style = MaterialTheme.typography.bodyMedium
                    )
                    Text(
                        text = "Zmiana powtórzeń: ${profile.repsChance}",
                        color = MaterialTheme.colorScheme.onBackground,
                        style = MaterialTheme.typography.bodyMedium
                    )
                    Text(
                        text = "Zmiana serii: ${profile.setsChance}",
                        color = MaterialTheme.colorScheme.onBackground,
                        style = MaterialTheme.typography.bodyMedium
                    )
                }

                Spacer(modifier = Modifier.height(16.dp))
            }

            Button(
                onClick = {
                    selectedProfile?.let { profile ->
                        coroutineScope.launch {
                            viewModel.setProfileForUser(profile.id)
                        }
                    }
                },
                modifier = Modifier
                    .fillMaxWidth()
                    .height(48.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = MaterialTheme.colorScheme.primary,
                    contentColor = MaterialTheme.colorScheme.onBackground
                ),
                shape = RoundedCornerShape(12.dp)
            ) {
                Text("Zapisz")
            }
        }
    }
}

File: iosApp/ui/ProfileScreen.swift
import SwiftUI

struct ProfileScreen: View {
    @ObservedObject var viewModel: ProfileViewModelWrapper
    @State private var selectedProfile: ProfileEntity?
    @State private var avatarUrl: String = ""
    
    var body: some View {
        ProfileContentView(
            viewModel: viewModel,
            selectedProfile: $selectedProfile,
            avatarUrl: avatarUrl,
            onAppear: {
                viewModel.loadData()
                let seed = Int.random(in: 1...1000)
                avatarUrl = "https://api.dicebear.com/7.x/personas/png?seed=user\(seed)"
            },
            onChangeProfile: { newValue in

            },
            onSave: {
                if let profile = selectedProfile {
                    viewModel.setProfile(profileId: profile.id)
                }
            }
        )
    }
}

struct ProfileContentView: View {
    let viewModel: ProfileViewModelWrapper
    @Binding var selectedProfile: ProfileEntity?
    let avatarUrl: String
    let onAppear: () -> Void
    let onChangeProfile: (ProfileEntity?) -> Void
    let onSave: () -> Void
    
    var body: some View {
        ScrollView {
            VStack(spacing: 20) {
                headerSection
                avatarSection
                userInfoSection
                Divider()
                profileSelectionSection
                profileDetailsSection
                saveButton
            }
            .padding()
        }
        .onAppear(perform: onAppear)
        .onChange(of: viewModel.selectedProfileId) { id in
            selectedProfile = viewModel.profiles.first(where: { $0.id == id })
        }
    }
    
    private var headerSection: some View {
        Text("Profil użytkownika")
            .font(.title)
            .bold()
    }
    
    private var avatarSection: some View {
        AsyncImage(url: URL(string: avatarUrl)) { image in
            image.resizable()
        } placeholder: {
            ProgressView()
        }
        .frame(width: 100, height: 100)
        .clipShape(Circle())
    }
    
    private var userInfoSection: some View {
        Group {
            Text(viewModel.username)
                .font(.title2)
            
            Text(viewModel.aiIdentifier)
                .foregroundColor(.gray)
                .font(.subheadline)
        }
    }
    
    private var profileSelectionSection: some View {
        VStack {
            Text("Wybierz profil")
                .font(.headline)
            
            Picker("Profil", selection: $selectedProfile) {
                ForEach(viewModel.profiles, id: \.id) { profile in
                    Text(profile.name).tag(Optional(profile))
                }
            }
            .pickerStyle(MenuPickerStyle())
            .onChange(of: selectedProfile, perform: onChangeProfile)
        }
    }
    
    private var profileDetailsSection: some View {
        Group {
            if let profile = selectedProfile {
                VStack(alignment: .leading, spacing: 8) {
                    Text("Szczegóły profilu:")
                        .font(.headline)
                        .foregroundColor(.accentColor)
                    
                    Text("Zmiana ciężaru: \(profile.weightChance, specifier: "%.1f") kg")
                    Text("Zmiana powtórzeń: \(profile.repsChance)")
                    Text("Zmiana serii: \(profile.setsChance)")
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(.top)
            }
        }
    }
    
    private var saveButton: some View {
        Button(action: onSave) {
            Text("Zapisz")
                .frame(maxWidth: .infinity)
        }
        .padding()
        .background(Color.accentColor)
        .foregroundColor(.white)
        .cornerRadius(10)
    }
}
